WordPress/gutenberg
# Post Comments block: hide the "Comments Closed" message

## Description
Solves https://github.com/WordPress/gutenberg/issues/35732. 

This PR changes the Post Comments rendering to hide the "Comments are closed" message when the following conditions are met:

- Comments are closed
- There are no comments to display
  
## How has this been tested?

1. Check out this PR: https://github.com/WordPress/twentytwentytwo/pull/127
2. Ensure "Allow comments" is disabled on the post, and no comments exist
3. Verify that no post comments markup is rendered
4. Go to a post with comments and verify they still appear

## Screenshots 

## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
